src/js/models/layerstructure.js

```diff
@@ -1213,13 +1212,19 @@ define(function (require, exports, module) {
      * @return {LayerStructure}
      */
     LayerStructure.prototype.setFillProperties = function (layerIDs, fillProperties) {
-        var emptyFill = new Fill(),
-            nextFill = emptyFill.setFillProperties(fillProperties),
-            nextLayers = Immutable.Map(layerIDs.reduce(function (map, layerID) {
-                return map.set(layerID, Immutable.Map({
-                    fill: nextFill
-                }));
-            }, new Map()));
+        var nextLayers = Immutable.Map(layerIDs.reduce(function (map, layerID) {
+            var layer = this.byID(layerID),
+                fill = layer.fill;
+
+            if (!fill) {
+                throw new Error("Unable to set fill properties: no fill for layer " + layer.name);
```

Comment:
Seems like it would be better to just make a new Fill here, but I guess that can wait for a later day.